digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -314,12 +314,18 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
       val projectLink = parsedBody.extract[NewRoadAddressExtractor]
       val roadLinks = projectService.getProjectRoadLinksByLinkIds(projectLink.linkIds)
       withDynTransaction {
-        val errorMessage = projectService.addNewLinksToProject(roadLinks, projectLink.projectId, projectLink.newRoadNumber, projectLink.newRoadPartNumber, projectLink.newTrackCode, projectLink.newDiscontinuity)
-        if (errorMessage == "") {
-          Map("success" -> true)
+        val elyUpdateErrorMessage = projectService.setProjectEly(projectLink.projectId, projectLink.roadEly)
```

Comment:
Let's not use strings and comparisons but Option[String] and None to mark success